keras/callbacks.py

```diff
@@ -399,7 +399,7 @@ def on_epoch_end(self, epoch, logs=None):
             self.epochs_since_last_save = 0
             filepath = self.filepath.format(epoch=epoch, **logs)
             if self.save_best_only:
-                current = logs.get(self.monitor)
+                current = logs.get(self.monitor)[-1]
```

Comment:
Not sure this is the behavior we actually want. Seems like a lot of hard-coded assumptions.